raiden-dapp/src/services/raiden-service.ts

```diff
@@ -314,6 +323,17 @@ export default class RaidenService {
   async getUDCCapacity(): Promise<BigNumber> {
     return this.raiden.getUDCCapacity();
   }
+
+  async getAvailability(address: string): Promise<boolean> {
+    try {
+      const { available } = await this.raiden.getAvailability(address);
+      return available;
+    } catch (e) {
+      this.store.commit('updatePresence', { [address]: false });
```

Comment:
If you store the whole presences object, and check for its presence as suggested above, you can just omit this, as object not here indicates presence is unknown, and this request can only ever fail on first `getAvailability` request, as following ones are always cached.
```suggestion
```

Reply:
I made this because of a `matrixPresenceUpdateFailed` exception (not sure of exact name right now) which gets thrown when a target is offline by the SDK

Reply:
This only happens if no user matching given address is found. Therefore, if address isn't in the Presences mapping, it may as well mean user is offline, so you can just swallow the error and not set anything.

Reply:
Then we would need to do the request again every time the user tries it with the same address, instead of having the info saved up in state. I guess the point of having the info in state was that the request does not happen every time?

I think it could be useful in the future as well, if we are going to persist state and display a "list of friends" or something like that, and we can quickly determine if everyone in the list is online or offline

Reply:
Well, they requesting again every time (the user inputs the address) is actually good, as it does a more complete search for candidate users. Again, it's not cached only if there was an error, which happens only if no user could be found matching given address (which shouldn't be common, and mostly indicate an address which didn't join yet or ever Raiden Network).

Better thinking, this isn't that crucial, I suggested more so we could keep the whole presences object (as I had suggested, but you rightfully argued it'd be better left for the future, when really needed), as constructing a "false" object is harder than just the boolean, so in that case it'd be better to just leave the entry out.
But, for the current state of the PR, this is not important. Thank you.